storage/spanstore/decorator/rate_limit.go

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package decorator
+
+import (
+	"errors"
+
+	"go.uber.org/ratelimit"
+
+	"github.com/jaegertracing/jaeger/model"
+	"github.com/jaegertracing/jaeger/storage/spanstore"
+)
+
+// UnlimitedRate is a value that may be passed to NewRateLimitedWriter to
+// indicate that no rate limiting should be applied.
+const UnlimitedRate = 0
+
+var errInvalidRate = errors.New("rate must be a non-negative integer")
+
+// NewRateLimitedWriter decorates a Writer with a rate limiter in order to limit
+// the number of writes per second.
+func NewRateLimitedWriter(writer spanstore.Writer, rate int) (spanstore.Writer, error) {
+	if rate < 0 {
+		return nil, errInvalidRate
+	}
+
+	var rateLimiter ratelimit.Limiter
+	if rate == UnlimitedRate {
+		rateLimiter = ratelimit.NewUnlimited()
+	} else {
+		rateLimiter = ratelimit.New(rate)
+	}
+	return &rateLimitedWriter{
+		w: writer,
+		l: rateLimiter,
+	}, nil
+}
+
+type rateLimitedWriter struct {
+	w spanstore.Writer
+	l ratelimit.Limiter
```

Comment:
Please use full field names. One letter names are ok for local variables when there is context.